vpr/src/route/rr_graph.cpp

```diff
@@ -2420,10 +2420,9 @@ static vtr::NdMatrix<std::vector<int>, 4> alloc_and_load_track_to_pin_lookup(vtr
 
     return track_to_pin_lookup;
 }
-
 /* TODO: This function should adapt RRNodeId */
-std::string describe_rr_node(int inode) {
-    auto& device_ctx = g_vpr_ctx.device();
+std::string describe_rr_node(int inode, const DeviceContext& device_ctx) {
```

Comment:
Actually, I prefer to have the ``DeviceContext`` as the 1st argument while the ``inode`` as the 2nd. In the argument list, usually we want to put big objects first and then followed by indices. Your call.